File: samples/inventory-service/inventory-py/inventory/stock.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class StockQuery:
    """Parameters for a stock-level lookup."""

    sku: str
    include_reserved: bool = False


@dataclass
class StockLevel:
    """Current stock counts for a single item."""

    sku: str
    available: int
    reserved: int
    last_updated: str

File: samples/inventory-service-with-predefined-models/inventory-go/types/stock.go
package types

// StockQuery contains the parameters for a stock-level lookup.
type StockQuery struct {
	SKU             string `json:"sku"`
	IncludeReserved bool   `json:"includeReserved,omitempty"`
}

// StockLevel holds the current counts for a single item.
type StockLevel struct {
	SKU         string `json:"sku"`
	Available   int    `json:"available"`
	Reserved    int    `json:"reserved"`
	LastUpdated string `json:"lastUpdated"`
}
